categories.go

```diff
@@ -27,18 +27,23 @@ import (
 )
 
 // categoriesHandler is a dynamic handler as it will also allow filtering in the future.
-func categoriesHandler(logger *zap.Logger, indexer Indexer, cacheTime time.Duration) func(w http.ResponseWriter, r *http.Request) {
-	return categoriesHandlerWithProxyMode(logger, indexer, proxymode.NoProxy(logger), cacheTime)
-}
-
-// categoriesHandler is a dynamic handler as it will also allow filtering in the future.
-func categoriesHandlerWithProxyMode(logger *zap.Logger, indexer Indexer, proxyMode *proxymode.ProxyMode, cacheTime time.Duration) func(w http.ResponseWriter, r *http.Request) {
+func categoriesHandler(logger *zap.Logger, options handlerOptions) (func(w http.ResponseWriter, r *http.Request), error) {
+	if options.proxyMode == nil {
+		logger.Warn("artifactsHandlerWithProxyMode called without proxy mode, defaulting to no proxy")
+		options.proxyMode = proxymode.NoProxy(logger)
+	}
+	if options.indexer == nil {
+		return nil, fmt.Errorf("indexer is required for categories handler")
+	}
+	if options.cacheTime == 0 {
+		return nil, fmt.Errorf("cache time must be set for categories handler")
+	}
```

Comment:
With this change of using `handlerOptions` struct, now each handler must check which options they require themselves and return error if there is anything unexpected.

In case of the proxy mode, if it is not set, I though to set the default proxy mode (NoProxy).

Reply:
As mentioned in https://github.com/elastic/package-registry/pull/1382#discussion_r2266846186, it has been updated `Enabled()` to check if struct pointer is nil and in that case it will be considered as disabled.